weblab/datasets/tests/test_views.py

```diff
@@ -120,6 +120,87 @@ def test_add_delete_add_file(self, logged_in_user, client, helpers, public_proto
                 assert fp.read() == file_map[f.name]
 
 
+@pytest.mark.django_db
+class TestDatasetDeletion:
+
+    def test_owner_can_delete_dataset(
+            self, helpers, logged_in_user, public_protocol , client
+    ):
+        helpers.add_permission(logged_in_user, 'create_dataset', Dataset)
+        dataset = recipes.dataset.make(author=logged_in_user, name='mydataset', protocol=public_protocol)
+        file_name = 'mydataset.csv'
+        file_contents = b'my test dataset'
+        recipes.dataset_file.make(  # Uploaded then deleted
```

Comment:
The comment here should probably be removed! And actually you could add `my_dataset_with_file` as an argument to the test (i.e. use that fixture) which lets you replace lines 129-145 by just `dataset = my_dataset_with_file`. And then you wouldn't need to use some other fixtures directly.

It might also be worth having a version of this test using `my_dataset` instead, to ensure users can delete 'partially created' datasets (i.e. without any files added yet).

Reply:
Done - thanks